nb2/__init__.py

```diff
@@ -6,10 +6,16 @@
 from flask_sqlalchemy import SQLAlchemy
 
 from config import DevelopmentConfig
-from nb2.bot.slackbot import SlackBot
 
 db = SQLAlchemy()
 migrate = Migrate()
+
+# This import needs to come after db since the import chain
+# in SlackBot eventually imports Person, which needs an
+# instantiated db.
+# TODO: Find a better way to handle circular imports?
+from nb2.bot.slackbot import SlackBot  # noqa
```

Comment:
Yea, I wonder if we need to decouple SlackBot and Person more? Haven't read to where this is an issue yet though so maybe it's just not possible.

Reply:
I'm remembering now as I'm reading through the SlackBot stuff, that I had initially thought Slackbot would just be responsible as an mediator between Slack and this app. So the SlackBot class would be responsible for managing the setup and sending messages and such, but would not be responsible for the logic involved in actually handling the command. 

So in that scenario, there might be some other CommandHandler class that would have method that handle the commands like `rememeber`  but has access to the bot so that it can handle commands and then just call `self.bot.send_message(result)`. 

I dunno what you think about this. It might help to make the bot class smaller, dumber and more decoupled from the rest of the logic.

Reply:
I'm definitely on board with moving the Actions into a service. As I was doing it though, I was concerned about the resulting architecture:
- (the new) `NostalgiaService` knows about `bot` (to send stuff)
- `slack_events` knows about `bot` (the decorator)
  - This one is the weird one; it only needs to know about bot because of the event adapter?
  - Can we move the event adapter to something else? What if we made slack_events a class? Then it can have the event adapter as a reference?
- The bot doesn't really know about anything else
- So it's like a triangle of sorts

Reply:
Going to make refactors in another ticket to make scopes clean!